zebra-chain/src/primitives/zcash_history.rs

```diff
@@ -233,14 +233,14 @@ impl Version for zcash_history::V1 {
     /// `orchard_root` is ignored.
     fn block_to_history_node(
         block: Arc<Block>,
-        network: Network,
+        network: &Network,
         sapling_root: &sapling::tree::Root,
         _orchard_root: &orchard::tree::Root,
     ) -> Self::NodeData {
         let height = block
             .coinbase_height()
             .expect("block must have coinbase height during contextual verification");
-        let network_upgrade = NetworkUpgrade::current(network, height);
+        let network_upgrade = NetworkUpgrade::current(&network, height);
```

Comment:
Looks like a double borrow.

```suggestion
        let network_upgrade = NetworkUpgrade::current(network, height);
```

Reply:
change applied